pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -934,6 +935,8 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 		// is done and there is no container to start.
 		if len(containersToStart) == 0 {
 			hasInitialized := false
+			// TODO: Remove this code path as logically it is the subset of the next
+			// code path.
```

Comment:
xref: https://github.com/kubernetes/kubernetes/pull/127162#discussion_r1923868666

Reply:
I'm not entirely sure about the intent behind keeping this code path. When will it be possible to remove it completely?

Reply:
we will need to introduce a separate feature gate so we switch logic safely. Keeping two code path doesn't really makes sense long term, but short term we want to be extra cautious